### PR TITLE
Merge/backport v5.0 intel audio v20190430

### PR DESCRIPTION
**Pull request by plbossart**
This backport is the first one that contains SOF patches merged upstream. Yay!

Only one minor change in the bytcht_es8316. 
I need two volunteers for a smoke test and approval.

### CODE REVIEW

**Comment by wenqingfu** [comment]
Congratulations!! 👍

**Comment by ranj063** [comment]
@plbossart now that SOF is upstream, what would be the working model for the incremental patches? 
Do we have to send it to alsa-devel after the PR is approved and patches merged in sof-dev?

**Comment by plbossart** [comment]
> @plbossart now that SOF is upstream, what would be the working model for the incremental patches?
> Do we have to send it to alsa-devel after the PR is approved and patches merged in sof-dev?

It's a good question. The first assumption is that all patches submitted on GitHub are upstream-worthy. That's not always the case, I just fixed checkpatch issues before sending the last batch... Also we may need to squash or reorder things to make the reviews easier, no reason e.g. to expose all the alsa community to the ABI changes or release updates. In addition, not everyone is familiar with sending stuff on the mailing list, and if there is feedback we will have to squash the different iterations somehow, or use reverts quite a bit.

In short, we should work to make sure all PRs are indeed incremental and professional-enough that they can go upstream without anyone (specifically me) in the loop, but we should keep plan A for now: submit patches through github and send to the ML directly only when things are fine or need direct feedback.

**Comment by plbossart** [comment]
@wenqingfu FWIW we have 23 patches submitted today to alsa-devel. I think we should probably have a discussion on whether we keep Haswell, Skylake/Kabylake, compressed, SPI and virtio, that's an additional 16 patches which have no future as is for now. It gives me work for no good reason and it's time to park them in an archive, or to make them fully supported features.

**Comment by kv2019i** [comment]
@plbossart I'm testing on Up2 as well, and that looks good (expected as Ranjani already tested). Playback with I2S and HDMI codecs is good, suspend works as expected. 

On BYT/QEMU (my other platform I have setup with upstream kernel), I'm getting an oops at boot:
```
[    1.942905] RIP: 0010:down_write+0x16/0x40
[...]
[    1.950906] Call Trace:
[    1.951381]  start_creating+0x59/0x100
[    1.951899]  __debugfs_create_file+0x2e/0x110
[    1.952442]  snd_sof_debugfs_io_item+0x7b/0x100 [snd_sof]
[    1.953025]  byt_fw_ready+0x2c9/0x3a0 [snd_sof_intel_byt]
[    1.953617]  ? irq_forced_thread_fn+0x70/0x70
[    1.954160]  snd_sof_ipc_msgs_rx+0x2e7/0x420 [snd_sof]
```

Not sure if this is related to your patches, will continue to investigate this.

**Comment by lyakh** [comment]
> @wenqingfu FWIW we have 23 patches submitted today to alsa-devel. I think we should probably have a discussion on whether we keep Haswell, Skylake/Kabylake, compressed, SPI and virtio, that's an additional 16 patches which have no future as is for now. It gives me work for no good reason and it's time to park them in an archive, or to make them fully supported features.

@plbossart I really wouldn't send SPI in the form that it is now in sof-dev, it's really useless like that.

**Comment by kv2019i** [comment]
@plbossart wrote:
> On BYT/QEMU (my other platform I have setup with upstream kernel), I'm getting an oops at boot:

Ok, rootcaused this. It seems we have a timing race in FW boot timing out. I'll address separately. I don't think this was added by your rebase, just a coincidence it happened now. So ok for the rebase.
